File: NET/API/Treehopper/PinMode.cs
namespace Treehopper
{
    /// <summary>
    /// An enumeration representing the different supported modes of this pin.
    /// </summary>
    public enum PinMode
    {
        /// <summary>
        /// Pin is reserved for other use
        /// </summary>
        Reserved,

        /// <summary>
        /// Pin is a digital input
        /// </summary>
        DigitalInput,

        /// <summary>
        /// Pin is a push-pull output
        /// </summary>
        PushPullOutput,

        /// <summary>
        /// Pin is an open-drain output
        /// </summary>
        OpenDrainOutput,

        /// <summary>
        /// Pin is an analog input
        /// </summary>
        AnalogInput,

        /// <summary>
        /// Pin is unassigned
        /// </summary>
        Unassigned
    }
}

File: Java/api/treehopper/src/main/java/io/treehopper/enums/PinMode.java
package io.treehopper.enums;

/**
 * Pin mode
 */
public enum PinMode {

    /**
     * Pin is reserved for other use
     */
    Reserved,

    /**
     * Pin is a digital input
     */
    DigitalInput,

    /**
     * Pin is a push-pull output
     */
    PushPullOutput,

    /**
     * Pin is an open-drain output
     */
    OpenDrainOutput,

    /**
     * Pin is an analog input
     */
    AnalogInput,

    /**
     * Pin is unassigned
     */
    Unassigned
}
